File: .github/helper/semgrep_rules/translate.js
// ruleid: frappe-translation-empty-string
__("")
// ruleid: frappe-translation-empty-string
__('')

// ok: frappe-translation-js-formatting
__('Welcome {0}, get started with ERPNext in just a few clicks.', [full_name]);

// ruleid: frappe-translation-js-formatting
__(`Welcome ${full_name}, get started with ERPNext in just a few clicks.`);

// ok: frappe-translation-js-formatting
__('This is fine');


// ok: frappe-translation-trailing-spaces
__('This is fine');

// ruleid: frappe-translation-trailing-spaces
__(' this is not ok ');
// ruleid: frappe-translation-trailing-spaces
__('this is not ok ');
// ruleid: frappe-translation-trailing-spaces
__(' this is not ok');

// ok: frappe-translation-js-splitting
__('You have {0} subscribers in your mailing list.', [subscribers.length])

// todoruleid: frappe-translation-js-splitting
__('You have') + subscribers.length + __('subscribers in your mailing list.')

// ruleid: frappe-translation-js-splitting
__('You have' + 'subscribers in your mailing list.')

// ruleid: frappe-translation-js-splitting
__('You have {0} subscribers' +
    'in your mailing list', [subscribers.length])

// ok: frappe-translation-js-splitting
__("Ctrl+Enter to add comment")

// ruleid: frappe-translation-js-splitting
__('You have {0} subscribers \
    in your mailing list', [subscribers.length])

File: .github/helper/semgrep_rules/translate.py
# Examples taken from https://frappeframework.com/docs/user/en/translations
# This file is used for testing the tests.

from frappe import _

full_name = "Jon Doe"
# ok: frappe-translation-python-formatting
_('Welcome {0}, get started with ERPNext in just a few clicks.').format(full_name)

# ruleid: frappe-translation-python-formatting
_('Welcome %s, get started with ERPNext in just a few clicks.' % full_name)
# ruleid: frappe-translation-python-formatting
_('Welcome %(name)s, get started with ERPNext in just a few clicks.' % {'name': full_name})

# ruleid: frappe-translation-python-formatting
_('Welcome {0}, get started with ERPNext in just a few clicks.'.format(full_name))


subscribers = ["Jon", "Doe"]
# ok: frappe-translation-python-formatting
_('You have {0} subscribers in your mailing list.').format(len(subscribers))

# ruleid: frappe-translation-python-splitting
_('You have') + len(subscribers) + _('subscribers in your mailing list.')

# ruleid: frappe-translation-python-splitting
_('You have {0} subscribers \
    in your mailing list').format(len(subscribers))

# ok: frappe-translation-python-splitting
_('You have {0} subscribers') \
    + 'in your mailing list'

# ruleid: frappe-translation-trailing-spaces
msg = _(" You have {0} pending invoice ")
# ruleid: frappe-translation-trailing-spaces
msg = _("You have {0} pending invoice ")
# ruleid: frappe-translation-trailing-spaces
msg = _(" You have {0} pending invoice")

# ok: frappe-translation-trailing-spaces
msg = ' ' + _("You have {0} pending invoices") + ' '

# ruleid: frappe-translation-python-formatting
_(f"can not format like this - {subscribers}")
# ruleid: frappe-translation-python-splitting
_(f"what" + f"this is also not cool")


# ruleid: frappe-translation-empty-string
_("")
# ruleid: frappe-translation-empty-string
_('')


class Test:
	# ok: frappe-translation-python-splitting
	def __init__(
			args
			):
		pass
